pkg/envfuncs/kind_funcs.go

```diff
@@ -28,31 +28,32 @@ import (
 	"sigs.k8s.io/e2e-framework/klient/wait/conditions"
 	"sigs.k8s.io/e2e-framework/pkg/env"
 	"sigs.k8s.io/e2e-framework/pkg/envconf"
+	"sigs.k8s.io/e2e-framework/support"
 	"sigs.k8s.io/e2e-framework/support/kind"
 )
 
-type kindContextKey string
+type clusterNameContextKey string
 
-// GetKindClusterFromContext helps extract the kind.Cluster object from the context.
+// GetClusterFromContext helps extract the kind.Cluster object from the context.
 // This can be used to setup and run tests of multi cluster kind.
-func GetKindClusterFromContext(ctx context.Context, clusterName string) (*kind.Cluster, bool) {
-	kindCluster := ctx.Value(kindContextKey(clusterName))
+func GetClusterFromContext(ctx context.Context, clusterName string) (support.E2EClusterProvider, bool) {
+	kindCluster := ctx.Value(clusterNameContextKey(clusterName))
 	if kindCluster == nil {
 		return nil, false
 	}
-	cluster, ok := kindCluster.(*kind.Cluster)
+	cluster, ok := kindCluster.(support.E2EClusterProvider)
 	return cluster, ok
 }
 
-// CreateKindCluster returns an env.Func that is used to
+// CreateCluster returns an env.Func that is used to
 // create a kind cluster that is then injected in the context
 // using the name as a key.
 //
 // NOTE: the returned function will update its env config with the
 // kubeconfig file for the config client.
-func CreateKindCluster(clusterName string) env.Func {
+func CreateCluster(clusterName string) env.Func {
```

Comment:
Is this renaming really required?  
As we are going to add other cluster providers (ex: kwok)  under this package it should also have all these functions with the provider name and I think it will ease user's understanding of which provider has been used to create the cluster.

Reply:
Yes and no. The reason behind refactoring this is to make this a bit more generic. That part is not yet sorted out. 

The Idea is `CreateCluster` will take a `ClusterProvider` type as argument and then fetch the provider using that name and then trigger the workflow from there. 

There will be bunch of alias handler created to `CreateKindCluster` that proxies the call to `CreateCluster` to retain compatibility with existing users. 

We can annotate those function as deprecated and slowly ask consumer to move towards the generic handlers across the board. Also the file will be renamed as `provider_funcs.go` may be so that there are not many direct references to kind on the code base